py-geopolars/python/geopolars/internals/geodataframe.py

```diff
@@ -1,23 +1,53 @@
-import polars
+from __future__ import annotations
 
+import polars as pl
+
+from geopolars import geopolars as core  # type: ignore
 from geopolars.internals.geoseries import GeoSeries
 
 DEFAULT_GEO_COLUMN_NAME = "geometry"
 
 
-class GeoDataFrame(polars.DataFrame):
+class GeoDataFrame(pl.DataFrame):
 
     _geometry_column_name = DEFAULT_GEO_COLUMN_NAME
 
     def __init__(self, data=None, columns=None, orient=None, *, geometry=None):
 
         # Wrap an existing polars DataFrame
-        if isinstance(data, polars.DataFrame):
+        if isinstance(data, pl.DataFrame):
             self._df = data._df
             return
 
         super().__init__(data, columns, orient)
 
+    def get_column(self, name: str) -> pl.Series | GeoSeries:
+        """
+        Get a single column as Series or GeoSeries by name. Return GeoSeries if requested column is geometry column.
+
+        Parameters
+        ----------
+        name : str
+            Name of the column to retrieve.
+
+        Examples
+        --------
+        >>> df = pl.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+        >>> df.get_column("foo")
+        shape: (3,)
+        Series: 'foo' [i64]
+        [
+                1
+                2
+                3
+        ]
+
+        """
+        series = super().get_column(name)
+        if name == self._geometry_column_name:
```

Comment:
👍 In the future, when polars supports extension types, we should be able to handle multiple geometry columns (and not forcing `_geometry_column_name`) by checking if the returned series is an extension type of type `geoarrow`

Reply:
sure, or have property returning names of geo columns. But for now i think it is enough to check this condition